CMakeLists.txt

```diff
@@ -210,6 +217,7 @@ if(NOT WIN32)
         core_optional_dep(LibDRM ENABLE_MIR)
       else()
         core_optional_dep(X ENABLE_X11)
+        core_optional_dep(XRandR ENABLE_X11)
```

Comment:
you have a point asking wheter rockchip is a platform or not. i was wondering too.

1. X11 platform
if the definition is that a platform defines its own windowing system than rockchip is not a platform here. this commits makes kodi run on latest linaro debian 9 distribution using ordinary X11 API and OpenGLES compliant API. the X server runs with glamor x11 display driver (hw accelerated OpenGLES rendering). this commit also uses the HW accelerator api of OpenGLES in X11 that you provide because it makes sense and just works.
the only essential issue is that instead of OpenGL only OpenGLES needs to be used and that seems something that current kodi builds arent familiar with in combination with X11 and XRandR. it seems strange that nobody experienced this before.

2. EGL WINDOWING PLATFORM
on the other hand, regarding the egl windowing system i now reuse AML logic for resolution selection just to avoid to implement an additional, similar one - its simply scanning standard linux framebuffers to get/set its modes. however the AML one offers much more that is simply not used for rockchip so it's a bit too much but its just there and already implemented.

3. EGL DRM WINDOWING
it would probably be best to implement a new variant here: however not dedicated for rockchip but for DRM in general as that would be the generic way for linux (could be even independent of X). in the end that would be best i guess. for playing videos it could offer a clean, generic way to get the frame from the hw decoder rendering it directly to drm planes / display without the need of any processing through GPU in between, all linux generic. by bypassing the GPU you would save memory bandwidth that is needed for 4K / 8K frames at high frame rates. however would be quite some effort.

anyway, right now i dont see a way to proceed so i guess for now we are stuck.